backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/constraints/UniqueTimeBoundariesDatesConstraintValidator.java

```diff
@@ -0,0 +1,18 @@
+package de.bund.digitalservice.ris.norms.adapter.input.restapi.constraints;
+
+import de.bund.digitalservice.ris.norms.adapter.input.restapi.schema.TimeBoundarySchema;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.HashSet;
+import java.util.List;
+
+/** Validates if TimeBoundary dates are unique */
+public class UniqueTimeBoundariesDatesConstraintValidator
+    implements ConstraintValidator<UniqueTimeBoundariesDatesConstraint, List<TimeBoundarySchema>> {
+  @Override
+  public boolean isValid(
+      List<TimeBoundarySchema> timeBoundaries, ConstraintValidatorContext context) {
+    return new HashSet<>(timeBoundaries.stream().map(TimeBoundarySchema::getDate).toList()).size()
```

Comment:
I love your approach @reckseba. It is using for sure a HashSet where when instantiating duplicates will automatically be removed but in my opinion it is not that readable.

What do you think about:

`return timeBoundaries.stream().map(TimeBoundarySchema::getDate).distinct().count() == timeBoundaries.size()`

This way we only rely on stream operations and don't need to create an intermediate in -memory data structure (HashSet)

But it is just my opinion, I don't think we won't ever had any performance issues because of this. It is for you to decide. Otherwise the rest is great!